source/ice_api.c

```diff
@@ -0,0 +1,1676 @@
+        {
+            stunRetStatus = StunSerializer_AddAttributePriority( &stunCxt,
+                                                                 pIceCandidatePair->pLocal->priority );
+        }
+
+        if( stunRetStatus == STUN_RESULT_OK )
+        {
+            stunRetStatus = StunSerializer_AddAttributeIceControlling( &stunCxt,
+                                                                       pIceAgent->tieBreaker );
+        }
+
+        if( stunRetStatus == STUN_RESULT_OK )
+        {
+            stunRetStatus = StunSerializer_AddAttributeUseCandidate( &stunCxt );
+        }
+
+        if( stunRetStatus == STUN_RESULT_OK )
+        {
+            memcpy( pIceCandidatePair->pTransactionIdStore,
+                    stunHeader.pTransactionId,
+                    STUN_HEADER_TRANSACTION_ID_LENGTH );
+
+            retStatus = Ice_PackageStunPacket( pIceAgent,
+                                               &stunCxt,
+                                               ( uint8_t * ) pIceAgent->remotePassword,
+                                               ( uint32_t ) strlen( pIceAgent->remotePassword ) * sizeof( char ),
+                                               pSendStunMessageBufferLength );
+        }
+
+        if( retStatus == ICE_RESULT_OK )
+        {
+            *ppSendStunMessageBuffer = &pIceAgent->stunMessageBuffer[ 0 ];
+        }
+    }
+
+    return retStatus;
+}
+
+/*------------------------------------------------------------------------------------------------------------------*/
+
+/* Ice_CreateRequestForConnectivityCheck - This API creates Stun Packet for connectivity check to the remote candidate . */
+
+IceResult_t Ice_CreateRequestForConnectivityCheck( IceAgent_t * pIceAgent,
+                                                   uint8_t ** ppSendStunMessageBuffer,
+                                                   uint32_t * pSendStunMessageBufferLength,
+                                                   uint8_t * pTransactionIdBuffer,
+                                                   IceCandidatePair_t * pIceCandidatePair )
+{
+    IceResult_t retStatus = ICE_RESULT_OK;
+    StunResult_t stunRetStatus = STUN_RESULT_OK;
+    StunContext_t stunCxt;
+    StunHeader_t stunHeader;
+
+    if( ( pIceAgent == NULL ) ||
+        ( ppSendStunMessageBuffer == NULL ) ||
+        ( pSendStunMessageBufferLength == NULL ) ||
+        ( pTransactionIdBuffer == NULL ) ||
+        ( pIceCandidatePair == NULL ) )
+    {
+        retStatus = ICE_RESULT_BAD_PARAM;
+    }
+
+    if( retStatus == ICE_RESULT_OK )
+    {
+        retStatus = Ice_InitializeStunPacket( pIceAgent,
+                                              &stunCxt,
+                                              pTransactionIdBuffer,
+                                              &pIceAgent->stunMessageBuffer[ 0 ],
+                                              &stunHeader,
+                                              1,
```

Comment:
Can we have #defines for these 1, 1 not sure what are those

Reply:
updated.